### ISSUE TITLE
Warn when the `style` attribute is missing a semicolon `;`

### ISSUE THREAD

**Issue by carson-katri** [liveview-native/live_view_native#182]
A common mistake is writing a `style` attribute with spaces as delimiters instead of semicolons `;`:

```heex
<Element style="padding() background(.red)" />
```

This should log a warning, ideally pointing to the position where the semicolon should be inserted. A message similar to this could be used:

```swift
Missing semicolon between modifiers in `style` attribute:

<Element style="padding() background(.red)" />
                         ^ add a semicolon after `padding()`

home_live.swiftui.neex:1:25
```

**Comment by bcardarella** [comment]
I just compared to web and in Chrome I don't see any support for whitespace deliminator between style rules so I'd like to not support that in ours.

But I agree some warning would be nice. @NduatiK I think we'd need to support that in the parser itself as currently I'm just doing `String.split(style, ";")`

**Comment by NduatiK** [comment]
Let's clarify:
- you want a warning when the parser finds rules that are only delimited by whitespaces
- the warning should only happen inside style attributes 
- warnings should be configurable from the caller, enabled on styles, disabled everywhere else (sheets)

@bcardarella, did I get that right?

**Comment by bcardarella** [comment]
so currently I split the value of the `style` attr on `;` and just iterate through each rule one by one and send it to the parser. What I think we should do instead is send the entire string to the parser. The current `RULES` parser is splitting on newline, but we should probably introduce the optional `;` to deliminate between rules:

```
padding(); background(.red)
```

should be the same as:

```
padding(); background(.red);
```

and

```
padding()
background(.red)
```

and

```
padding();
background(.red);
```

Does that make sense?

**Comment by NduatiK** [comment]
Makes sense.
One more clarification, would a missing semicolon be expected on sheets and produce a warning?

**Comment by bcardarella** [comment]
are you asking if the following:

```
padding();
fooBar()
barBaz();
```

where the 2nd rule has no semicolor?

**Comment by NduatiK** [comment]
Yes, should the SHEET sigil produce a warning if the semicolon is missing?

**Comment by bcardarella** [comment]
That's a good question, are semicolons required for multiline in css or not? I forget what the current state of that is. If it is then we should probably just follow that but it may need to be a v0.4 thing and accept the breaking change

**Comment by cmnstmntmn** [comment]
Going to link this issue https://github.com/liveview-native/liveview-client-swiftui/issues/1421, since they might use the same parser internally.
